Source/Diagnostics/ComputeDiagFunctors/DivBFunctor.cpp

```diff
@@ -20,8 +22,32 @@ DivBFunctor::operator()(amrex::MultiFab& mf_dst, int dcomp, const int /*i_buffer
     constexpr int ng = 1;
     // A cell-centered divB multifab spanning the entire domain is generated
     // and divB is computed on the cell-center, with ng=1.
-    amrex::MultiFab divB( warpx.boxArray(m_lev), warpx.DistributionMap(m_lev), 1, ng );
+    amrex::MultiFab divB( warpx.boxArray(m_lev), warpx.DistributionMap(m_lev), warpx.ncomps, ng );
     warpx.ComputeDivB(divB, 0, m_arr_mf_src, WarpX::CellSize(m_lev) );
-    // Coarsen and Interpolate from divB to coarsened/reduced_domain mf_dst
+    // // Coarsen and Interpolate from divB to coarsened/reduced_domain mf_dst
+    // CoarsenIO::Coarsen( mf_dst, divB, dcomp, 0, nComp(), 0, m_crse_ratio);
+#ifdef WARPX_DIM_RZ
+    if (m_convertRZmodes2cartesian) {
+        // In cylindrical geometry, sum real part of all modes of divE in
+        // temporary multifab mf_dst_stag, and cell-center it to mf_dst.
+        AMREX_ALWAYS_ASSERT_WITH_MESSAGE(
+            nComp()==1,
+            "The RZ averaging over modes must write into 1 single component");
+        amrex::MultiFab mf_dst_stag(divB.boxArray(), warpx.DistributionMap(m_lev), 1, divB.nGrowVect());
+        // Mode 0
+        amrex::MultiFab::Copy(mf_dst_stag, divB, 0, 0, 1, divB.nGrowVect());
+        for (int ic=1 ; ic < divB.nComp() ; ic += 2) {
+            // Real part of all modes > 0
+            amrex::MultiFab::Add(mf_dst_stag, divB, ic, 0, 1, divB.nGrowVect());
+        }
+        CoarsenIO::Coarsen( mf_dst, mf_dst_stag, dcomp, 0, nComp(), 0,  m_crse_ratio);
+    } else {
+        CoarsenIO::Coarsen( mf_dst, divB, dcomp, 0, nComp(), 0, m_crse_ratio);
```

Comment:
Just add as a todo as discussed today, exists already in old code:
```suggestion
        // TODO check if coarsening is needed, otherwise copy
        CoarsenIO::Coarsen( mf_dst, divB, dcomp, 0, nComp(), 0, m_crse_ratio);
```

Reply:
TODO'd!